### PR TITLE
Port BGPFilter changes to CC and some fixes

### PR DESCRIPTION
**Pull request by Josh-L**
Copy BGPFilter documentation from CE to CC, fix a small typo in example BGPFilter resource, and fix sidebar navigation by adding a reference to new BGPFilter resource.

### CODE REVIEW

**Comment by ctauchen** [comment]
@Josh-Tigera Does this apply also to CC 3.16?

**Comment by Josh-L** [comment]
> @Josh-Tigera Does this apply also to CC 3.16?


@ctauchen good catch it does yes

**Comment by Josh-L** [comment]
I don't see a 3.16 directory in the CE versioned docs

**Comment by ctauchen** [comment]
> I don't see a 3.16 directory in the CE versioned docs

CE 3.16 and CC 3.16 are unreleased, what we call 'next'. To modify files for those versions, you make changes to the unversioned directories:

./calico-enterprise/
./calico-cloud/

When we publish docs on release day, we basically copy those directories, give them a version number, and move them to their respective versioned docs directory.

So right now, you've prepared a change to the CE 3.16 docs. 
If this change belongs in CC 3.16, we'd have to make the corresponding change there.

**Comment by Josh-L** [comment]
> > I don't see a 3.16 directory in the CE versioned docs
> 
> CE 3.16 and CC 3.16 are unreleased, what we call 'next'. To modify files for those versions, you make changes to the unversioned directories:
> 
> ./calico-enterprise/ ./calico-cloud/
> 
> When we publish docs on release day, we basically copy those directories, give them a version number, and move them to their respective versioned docs directory.
> 
> So right now, you've prepared a change to the CE 3.16 docs. If this change belongs in CC 3.16, we'd have to make the corresponding change there.

This isn't a CC-specific change, but I'll admit I know almost nothing about CC so I don't know whether I need to commit there or not. For the features that overlap between CC/CE are the docs duplicated between the directories?

File: calico-cloud/networking/configuring/bgp.mdx [hunk 1]
```diff
@@ -256,8 +259,106 @@ You can configure an AS for a particular node by modifying the node object using
...
+    - action: Accept
+      matchOperator: NotEqual
+      cidr: 5000::0/64
+```
+
+### Configure a BGP peer with a BGP filter
+
+BGP peers can use BGP filters to control which routes are imported or exported between them.
+
+The following example creates a BGPFilter and associates it with a BGPPeer
```
Comment (ctauchen) [inline calico-cloud/networking/configuring/bgp.mdx:296]: Please add a blank line before the `:::note`.

**Review by lwr20 — APPROVED** [review]
LGTM!

**Review by ctauchen — APPROVED** [review]

**Pull request merged**